### PR TITLE
feat: Add ID to interactive element

### PR DESCRIPTION
**Pull request by BisuGit**
## Description
added id to interactive elements. 

## DevQA
click on the interactive element and see that it is different from the text. 

### DevQA Prep

- Run `pod install`
- Test `RichTextView` to ensure that it passes (unit and UI tests)
- Navigate to the `Example` project in the root `Example` folder and run the app
- Ensure that everything in the `Example` app looks visually correct

### CODE REVIEW

File: Example/Source/ViewController.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ class ViewController: UIViewController {
             InputOutputModuleView(text: "<a href='https://www.google.com'>jump to page</a>"),
             InputOutputModuleView(text: Text.tableHTML),
             InputOutputModuleView(text: "<blockquote>Here is a blockquote</blockquote>"),
-            InputOutputModuleView(text: "Look [interactive-element]This is an interactive element[/interactive-element] Wow")
+            InputOutputModuleView(text: "Look [interactive-element id = 123]This is an interactive element[/interactive-element] Wow")
```
Comment (aelkady) [inline Example/Source/ViewController.swift]: Can we make the example `id=123` (no spaces) and make sure the regex is in the same way

File: Source/Text Parsing/RichTextParser.swift [hunk 2]
```diff
@@ -14,7 +14,8 @@ class RichTextParser {
         static let mathTagName = "math"
         static let interactiveElementTagName = "interactive-element"
         static let latexRegex = "\\[\(ParserConstants.mathTagName)\\](.*?)\\[\\/\(ParserConstants.mathTagName)\\]"
-        static let interactiveElementRegex = "\\[\(ParserConstants.interactiveElementTagName)\\](.*?)\\[\\/\(ParserConstants.interactiveElementTagName)\\]"
+        static let interactiveElementRegex =
```
Comment (aelkady) [inline Source/Text Parsing/RichTextParser.swift]: If it is too big to fit on one line would prefer you do
```
static let interactiveElementRegex = """
\\[\(ParserConstants.interactiveElementTagName)\\s+id\\s*=\\s*.+\\].*\\[\\/\(ParserConstants.interactiveElementTagName)\\]
"""

File: Source/Text Parsing/RichTextParser.swift [hunk 3]
```diff
@@ -145,8 +146,9 @@ class RichTextParser {
 
     func extractInteractiveElement(from input: String) -> NSMutableAttributedString {
         let interactiveElementTagName = ParserConstants.interactiveElementTagName
-        let interactiveElementText = input.getSubstring(inBetween: "[\(interactiveElementTagName)]", and: "[/\(interactiveElementTagName)]") ?? input
-        let attributes: [NSAttributedString.Key: Any] = [.customLink: interactiveElementText, .foregroundColor: self.interactiveTextColor, .font: self.font]
+        let interactiveElementID = input.getSubstring(inBetween: "[\(interactiveElementTagName) id = ", and: "]") ?? input
```
Comment (aelkady) [inline Source/Text Parsing/RichTextParser.swift]: Can you change this check to remove the spaces, so `id=`?
An example of what the regex should be catching is
`[interactive-element id=123]Hey![/interactive-element]`

**Comment by tophat-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">
<p>If anything other than <code>RichTextView</code> appears in the code coverage section please add it<br />
to the <code>exclude_targets</code> property of the <code>xcov.report()</code> function in the Dangerfile</p>
</td>
    </tr>
  </tbody>
</table>

## Current coverage for RichTextView.framework is `87.09%`
Files changed | - | - 
--- | --- | ---
UITextViewGenerator.swift | `47.62%` | :no_entry_sign:
RichTextParser.swift | `93.20%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**